Assets/Scripts/EnemyBehaviour.cs

```diff
@@ -0,0 +1,180 @@
+    aware = 1,
+    /// <summary>
+    /// Enemy is engaged in conflict with player
+    /// </summary>
+    engaged = 2
+}
+
+/// <summary>
+/// Exception thrown in the event that an ENEMY_STATE value is incompatible with a function
+/// </summary>
+public class InvalidEnemyStateException : System.ArgumentException {
+    /// <summary>
+    /// Throws an exception with a given error message.
+    /// </summary>
+    /// <param name="message"> Error message </param>
+    public InvalidEnemyStateException(string message) : base(message) { }
+}
+
+/// <summary>
+/// Abstract class providing framework for Enemy actions
+/// </summary>
+public abstract class EnemyBehaviour {
+    /// <summary>
+    /// Array storing information about intended use of the EnemyBehaviour in different ENEMY_STATEs
+    /// </summary>
+    /// <remarks>
+    /// Given an ENEMY_STATE state, has the property that _use_case[(int)state] iff the EnemyBehaviour is designed for use with state.
+    /// An EnemyBehaviour is not designed for use with any state by default, so _use_case is initialized as all false.
+    /// </remarks>
+    protected bool[] _use_case = new bool[3];
+
+    /// <summary>
+    /// Enemy's Rigidbody2D
+    /// </summary>
+    protected Rigidbody2D _enemyBody;
+
+    /// <summary>
+    /// Checks if an ENEMY_STATE is an intended use case for the EnemyBehaviour
+    /// </summary>
+    /// <param name="state"> The ENEMY_STATE in which the EnemyBehaviour will be used by an Enemy. Can be set to null to skip check. </param>
+    /// <exception cref="InvalidEnemyStateException"> Thrown if state is not an intended use case for the EnemyBehaviour </exception>
+    protected void _CheckUseCase(ENEMY_STATE? state) {
+        if(
+            state is ENEMY_STATE state_val
+            && !this._use_case[(int)state_val]
+        ) {
+            throw new InvalidEnemyStateException("INVALID ENEMY STATE ERROR: " + this.GetType() + " does not support " + state_val.ToString() + " enemy state.");
+        }
+    }
+
+    /// <summary>
+    /// Performs the next frame of action required for an EnemyBehaviour
+    /// </summary>
+    /// <returns> Whether behaviour can be interrupted on the next frame by a change of ENEMY_STATE </returns>
+    public abstract bool Act();
+}
+
+/// <summary>
+/// EnemyBehaviour in which Enemy does not move or perform any action
+/// </summary>
+/// <remarks>
+/// Intended for use in idle and aware ENEMY_STATEs
+/// </remarks>
+public class DoNothing : EnemyBehaviour {
+    /// <summary>
+    /// Constructs a new DoNothing behaviour
+    /// </summary>
+    /// <param name="enemy"> Enemy who will use the DoNothing behaviour </param>
+    /// <param name="state"> ENEMY_STATE in which the DoNothing behaviour will be used. Null to skip use-case checking </param>
+    public DoNothing(Enemy enemy, ENEMY_STATE? state) {
+        this._use_case[(int)ENEMY_STATE.idle] = true;
+        this._use_case[(int)ENEMY_STATE.aware] = true;
```

Comment:
May we abstract into a small helper with the following signature: `this._set_use_cases(case1, case2, ...)` or alternatively `this._set_use_cases([case1, case2, ...])`

Reply:
Absolutely. Done.